src/ipc/ipc-common/IPCProxy.h

```diff
@@ -61,45 +61,45 @@ class IPCProxy : public WrapperType, public IPCProxyNewBase
     {
         QObject::connect(ipc(), &IPCProxyBinderBase::complete, this, [this] () {
                 m_localProviderBinder.init();
-                if (m_localProviderBinder.provider() == nullptr) {
-                    auto serviceName = ipc()->serviceName();
-                    auto objPath = ipc()->objectPath();
-                    for (auto& proxy : m_ipcProxies) {
-                        auto proxyAdapterIPCBinder = proxy.ipcBinder;
-                        if (proxyAdapterIPCBinder != nullptr) {
-                            proxyAdapterIPCBinder->setObjectPath(objPath);
-                            if(!serviceName.isEmpty()) {
-                                proxyAdapterIPCBinder->setServiceName(serviceName);
-                            }
-                            proxyAdapterIPCBinder->connectToServer();
-                            QObject::connect(proxyAdapterIPCBinder, &IPCProxyBinderBase::serviceAvailableChanged, this, &IPCProxy::refreshProvider);
-                        }
+                for (auto& proxy : m_ipcProxies) {
+                    auto ipcBinder = proxy.ipcBinder;
+                    if (ipcBinder != nullptr) {
+                        QObject::connect(ipcBinder, &IPCProxyBinderBase::serviceAvailableChanged, this, &IPCProxy::refreshProvider);
                     }
                 }
                 this->refreshProvider();
-            });
-
+        });
     }
 
     void refreshProvider() override
```

Comment:
this function breaks SRP now https://en.wikipedia.org/wiki/Single-responsibility_principle
Please decouple responsibilities.
This function is also getting too big/too long. Please split into smaller pieces.

Reply:
Introduced sub-method